iceoryx_posh/include/iceoryx_posh/roudi/roudi_app.hpp

```diff
@@ -78,19 +70,18 @@ class RouDiApp
     bool waitForSignal() const noexcept;
 
     iox::log::LogLevel m_logLevel{iox::log::LogLevel::kWarn};
-    config::MonitoringMode m_monitoringMode{config::MonitoringMode::ON};
+    roudi::MonitoringMode m_monitoringMode{roudi::MonitoringMode::ON};
     bool m_run{true};
     RouDiConfig_t m_config;
 
-    posix::Semaphore m_semaphore = std::move(posix::Semaphore::create(posix::CreateUnnamedSingleProcessSemaphore, 0u)
-                                                 .or_else([](posix::SemaphoreError&) {
-                                                     std::cerr << "Unable to create the semaphore for RouDi"
-                                                               << std::endl;
-                                                     std::terminate();
-                                                 })
-                                                 .value());
+    posix::Semaphore m_semaphore =
+        std::move(posix::Semaphore::create(posix::CreateUnnamedSingleProcessSemaphore, 0u)
+                      .or_else([](posix::SemaphoreError&) {
+                          errorHandler(Error::kROUDI_APP__FAILED_TO_CREATE_SEMAPHORE, nullptr, ErrorLevel::FATAL);
```

Comment:
After calling the error handler we have to explicitly terminate. If we don't do this and someone configures the error handling so that they ignore this we run completely into the undefined behavior land. 

This class is unusable when the semaphore acquire fails. The usual way to handle this kind of problem would be the `Creation` design pattern ... or terminate.

Reply:
Hmm, that seems like an unreasonable exception to me :smiley: If the user is configuring the error handling wrong, than it is not our fault. We need to have a single place where we can catch all the errors, this being the error handler. IIRC we wanted to remove `std::terminate` occurrences except for the classes that need to run independent from `iceoryx_utils` (see #261) e.g. `cxx::vector`

@MatthiasKillat Can you confirm? Did not get a grasp on your error handling document yet.

Reply:
The same as in (EH) applies here in my opinion. An error is propagated and at this point we decide that without a semaphore we cannot continue and hence handle this error as a fatal error. An additional terminate would be superfluous. 
Writing a better error handler belongs to a different issue.